src/ynEIGEN/TokenStakingNode.sol

```diff
@@ -162,77 +157,124 @@ contract TokenStakingNode is
     }
 
     /**
-     * @notice Completes queued withdrawals for a specific strategy
-     * @param _nonce The nonce of the withdrawal
-     * @param _startBlock The block number when the withdrawal was queued
-     * @param _shares The number of shares to withdraw
-     * @param _strategy The strategy from which to withdraw
-     * @param _middlewareTimesIndexes The indexes of middleware times to use for the withdrawal
+     * @notice Completes queued withdrawals with receiveAsTokens set to true
+     * @param withdrawals Array of withdrawals to complete
+     * @param middlewareTimesIndexes Array of middleware times indexes
      * @param updateTokenStakingNodesBalances If true calls updateTokenStakingNodesBalances for yieldNestStrategyManager
      */
     function completeQueuedWithdrawals(
-        uint256 _nonce,
-        uint32 _startBlock,
-        uint256 _shares,
-        IStrategy _strategy,
-        uint256[] memory _middlewareTimesIndexes,
+        IDelegationManager.Withdrawal[] memory withdrawals,
+        uint256[] memory middlewareTimesIndexes,
         bool updateTokenStakingNodesBalances
-    ) public onlyTokenStakingNodesWithdrawer {
+    ) public onlyTokenStakingNodesWithdrawer onlyWhenSynchronized {
+
+        if (withdrawals.length != middlewareTimesIndexes.length) {
+            revert ArrayLengthMismatch();
+        }
 
         IDelegationManager _delegationManager = tokenStakingNodesManager.delegationManager();
+        uint256[] memory _balancesBefore = new uint256[](withdrawals.length);
+        IERC20[][] memory _tokens = new IERC20[][](withdrawals.length);
+        IStrategy[] memory _strategies = new IStrategy[](withdrawals.length);
+        bool[] memory _receiveAsTokens = new bool[](withdrawals.length);
+        IWrapper _wrapper = IYieldNestStrategyManager(tokenStakingNodesManager.yieldNestStrategyManager()).wrapper();
 
-        IDelegationManager.Withdrawal[] memory _withdrawals = new IDelegationManager.Withdrawal[](1);
-        {
-            IStrategy[] memory _strategiesArray = new IStrategy[](1);
-            _strategiesArray[0] = _strategy;
-            uint256[] memory _sharesArray = new uint256[](1);
-            _sharesArray[0] = _shares;
-            _withdrawals[0] = IDelegationManager.Withdrawal({
-                staker: address(this),
-                delegatedTo: _delegationManager.delegatedTo(address(this)),
-                withdrawer: address(this),
-                nonce: _nonce,
-                startBlock: _startBlock,
-                strategies: _strategiesArray,
-                shares: _sharesArray
-            });
+        for (uint256 i = 0; i < withdrawals.length; i++) {
+            IStrategy _strategy = withdrawals[i].strategies[0];
```

Comment:
same issue here with the fact that it's implied and not asserted that there's exactly one strategy in the withdrawal

this may be false.

We must assert and not process if that is the case